### PR TITLE
`PurchasesStoreProduct`: fixed inconsistent snake_case properties

### LINKED ISSUE #388: [5.0.0-beta.1] restorePurchases doesn't work
Probably because it hasn't been [renamed here](https://github.com/RevenueCat/react-native-purchases/blob/5.0.0-beta.1/src/purchases.ts#L434)

As an aside, the mix of `snake_case` and `camelCase` in 5.0 is really annoying. ex: `offering.product.intro_price.periodNumberOfUnits`

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #388.
As far as I know this was just a mistake in version 4.x, so the 5.0 major bump is a good opportunity to fix this.

Depends on https://github.com/RevenueCat/purchases-hybrid-common/pull/187

### CODE REVIEW

**Comment by NachoSoto** [comment]
Stand by, there are tests that need updating too.

File: scripts/setupJest.js [hunk 1]
```diff
@@ -87,15 +81,17 @@ global.offeringsStub = {
     twoMonth: {
       offeringIdentifier: 'default',
       product: {
-        intro_price_period_number_of_units: 16,
-        intro_price_period_unit: 'DAY',
-        intro_price_cycles: 1,
-        intro_price_period: 'P2W2D',
-        intro_price_string: '$0.00',
-        intro_price: 0,
+        introPrice: {
```
Comment (NachoSoto) [inline scripts/setupJest.js:84]: These were all out of date too. We didn't catch it because tests only do a raw comparison.

File: Gemfile.lock [hunk 2]
```diff
@@ -275,10 +275,11 @@ GEM
...
   arm64-darwin-21
+  universal-darwin-22
 
 DEPENDENCIES
   cocoapods
   fastlane
 
 BUNDLED WITH
-   2.3.14
+   2.3.6
```
Comment (NachoSoto) [inline Gemfile.lock:285]:
```suggestion
   2.3.14
```

File: Gemfile.lock [hunk 3]
```diff
@@ -275,10 +275,11 @@ GEM
 
 PLATFORMS
   arm64-darwin-21
+  universal-darwin-22
```
Comment (NachoSoto) [inline Gemfile.lock:278]:
```suggestion
```

File: VERSIONS.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 | Version | iOS version | Android version | Common files version |
 |---------|-------------|-----------------|----------------------|
+| 5.0.0-beta.2  | 4.9.1 | 5.3.0           | 4.4.0                |
```
Comment (NachoSoto) [inline VERSIONS.md:3]:
```suggestion
| 5.0.0-beta.2  | 4.9.1 | 5.3.0           | 4.0.0                |
```

**Comment by aboedo** [comment]
> As far as I know this was just a mistake in version 4.x, so the 5.0 major bump is a good opportunity to fix this.

I _think_ the naming was a mistake in like a 2.x, kind of as a hack to get out of the `introPrice` property and into an object

**Review by aboedo — APPROVED** [review]
This is a great cleanup :chefskiss:

**Pull request merged**